presto-main/src/test/java/com/facebook/presto/operator/TestUncheckedByteArrays.java

```diff
@@ -0,0 +1,228 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.google.common.primitives.Bytes;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+
+import static com.facebook.presto.operator.UncheckedByteArrays.fill;
+import static com.facebook.presto.operator.UncheckedByteArrays.getByte;
+import static com.facebook.presto.operator.UncheckedByteArrays.getInt;
+import static com.facebook.presto.operator.UncheckedByteArrays.getLong;
+import static com.facebook.presto.operator.UncheckedByteArrays.getShort;
+import static com.facebook.presto.operator.UncheckedByteArrays.setByte;
+import static com.facebook.presto.operator.UncheckedByteArrays.setBytes;
+import static com.facebook.presto.operator.UncheckedByteArrays.setInt;
+import static com.facebook.presto.operator.UncheckedByteArrays.setInts;
+import static com.facebook.presto.operator.UncheckedByteArrays.setLong;
+import static com.facebook.presto.operator.UncheckedByteArrays.setShort;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static sun.misc.Unsafe.ARRAY_BYTE_INDEX_SCALE;
+import static sun.misc.Unsafe.ARRAY_INT_INDEX_SCALE;
+import static sun.misc.Unsafe.ARRAY_LONG_INDEX_SCALE;
+import static sun.misc.Unsafe.ARRAY_SHORT_INDEX_SCALE;
+
+public class TestUncheckedByteArrays
+{
+    private static final int POSITIONS_PER_PAGE = 10000;
+    private static final Random RANDOM = new Random(0);
+
+    private TestUncheckedByteArrays()
+    {}
+
+    @Test
+    public static void testSetByte()
+    {
+        final byte[] destination = new byte[POSITIONS_PER_PAGE];
+
+        byte[] sourceBytes = new byte[POSITIONS_PER_PAGE];
+        RANDOM.nextBytes(sourceBytes);
+        List<Byte> source = Bytes.asList(sourceBytes);
+        testSequential(
+                source,
+                ARRAY_BYTE_INDEX_SCALE,
+                destination,
+                (bytes, index, value) -> setByte((byte[]) bytes, (int) index, (byte) value),
+                (bytes, index) -> getByte((byte[]) bytes, (int) index));
+
+        source = new ArrayList<Byte>(Collections.nCopies(POSITIONS_PER_PAGE, (byte) 0));
+        testRandom(
+                source,
+                ARRAY_BYTE_INDEX_SCALE,
+                destination,
+                (bytes, index, value) -> setByte((byte[]) bytes, (int) index, (byte) value),
```

Comment:
- it is not necessary to pass in `destination` parameter into `testSequential` and `testRandom`; these methods can create a local variable themselves
- "source should contain identical elements." comment on `testRandom` suggests that the signature of the method should be different, e.g. take a single source value instead of a list
- BiFunction and TriFunction generics needs to have their parameters specified; this would allow simpler calling code

```
    @Test
    public static void testSetByte()
    {
        byte[] source = new byte[POSITIONS_PER_PAGE];
        RANDOM.nextBytes(source);
        testSequential(
                Bytes.asList(source),
                ARRAY_BYTE_INDEX_SCALE,
                UncheckedByteArrays::setByte,
                UncheckedByteArrays::getByte);

        testRandom(
                (byte) 0,
                ARRAY_BYTE_INDEX_SCALE,
                UncheckedByteArrays::setByte,
                UncheckedByteArrays::getByte);
    }

...
    private static <T> void testSequential(List<T> source, int elementSize, TriFunction<byte[], Integer, T> writeFunction, BiFunction<byte[], Integer, T> readFunction)
    {
        byte[] destination = new byte[source.size() * elementSize];

        int index = 0;
        for (T value : source) {
            index = writeFunction.apply(destination, index, value);
        }
        assertEquals(index, POSITIONS_PER_PAGE * elementSize);
        assertCopied(source, destination, elementSize, readFunction);
    }

    private static <T> void testRandom(T sourceValue, int elementSize, TriFunction<byte[], Integer, T> writeFunction, BiFunction<byte[], Integer, T> readFunction)
    {
        byte[] destination = new byte[POSITIONS_PER_PAGE * elementSize];
        Arrays.fill(destination, (byte) 0);

        int index = 0;
        for (int i = 0; i < POSITIONS_PER_PAGE; i++) {
            int newIndex = writeFunction.apply(destination, index, sourceValue);
            if (i % 2 == 0) {
                index = newIndex;
            }
        }

        assertEquals(index, POSITIONS_PER_PAGE / 2 * elementSize);
        assertCopied(nCopies(POSITIONS_PER_PAGE / 2, sourceValue), Arrays.copyOf(destination, POSITIONS_PER_PAGE / 2), elementSize, readFunction);
        assertCopied(nCopies(POSITIONS_PER_PAGE / 2, sourceValue), Arrays.copyOf(destination, POSITIONS_PER_PAGE / 2), elementSize, readFunction);
    }
```